src/utils/AdaptiveTimeStep.cpp

```diff
@@ -32,6 +32,22 @@ AdaptiveTimeStep::AdaptiveTimeStep ()
         queryWithParser(ppa, "nt_per_betatron", m_nt_per_betatron);
     }
     DeprecatedInput("hipace", "do_adaptive_time_step", "dt = adaptive");
+
+    // get number of beams
+    amrex::Vector<std::string> beam_names;
+    amrex::ParmParse ppb("beams");
+    getWithParser(ppb, "names", beam_names);
+    if (beam_names[0] == "no_beam") return;
+    const int nbeams = beam_names.size();
+
```

Comment:
When the `AdaptiveTimeStep` constructor is called, I believe `m_multi_beam` already has the number of beams. Could we just pass `m_multi_beam.get_nbeams()` to the `AdaptiveTimeStep` constructor? Would avoid duplication. We may even store it in a member variable of `AdaptiveTimeStep`, as we use it in several places.

Reply:
Yep, that worked, thanks!